impl/deps/postgres/instant/instant.go

```diff
@@ -0,0 +1,218 @@
+	"kantoku/common/data/transactional"
+	"kantoku/impl/deps/postgres"
+	"kantoku/unused/backend/framework/depot/deps"
+)
+
+var _ postgres.Deps = &Deps{}
+
+type Deps struct {
+	resolvedGroups pool.Pool[string]
+	client         *pgxpool.Pool
+}
+
+// New creates an instance of the dependency manager with *instant* resolving.
+// client is a connection to a postgres database (all dependencies and groups are stored there)
+// queue is for storing resolved groups waiting to be processed
+func New(client *pgxpool.Pool, groupsQueue pool.Pool[string]) *Deps {
+	return &Deps{
+		resolvedGroups: groupsQueue,
+		client:         client,
+	}
+}
+
+func (d *Deps) InitTables(ctx context.Context) error {
+	sql := `
+		CREATE TABLE InstantDependencies (
+			id varchar(255),
+			resolved bool,
+			PRIMARY KEY (id)
+		);
+		
+		CREATE TABLE InstantGroups (
+			id varchar(255),
+			pending int,
+			PRIMARY KEY (id)
+		);
+		
+		CREATE TABLE InstantGroupDependencies (
+			dependency_id varchar(255),
+			group_id varchar(255),
+			PRIMARY KEY (dependency_id, group_id),
+			CONSTRAINT fk_group
+			  FOREIGN KEY(group_id) 
+			  REFERENCES InstantGroups(id)
+		      ON DELETE CASCADE
+		);
+	`
+	_, err := d.client.Exec(ctx, sql)
+	return err
+}
+
+func (d *Deps) Dependency(ctx context.Context, id string) (deps.Dependency, error) {
+	sql := `SELECT id, resolved FROM InstantDependencies WHERE id = $1`
+	row := d.client.QueryRow(ctx, sql, id)
+
+	var model deps.Dependency
+	if err := row.Scan(&model.ID, &model.Resolved); err != nil {
+		return deps.Dependency{}, err
+	}
+	return model, nil
+}
+
+func (d *Deps) DropTables(ctx context.Context) error {
+	sql := `DROP TABLE InstantDependencies, InstantGroups, InstantGroupDependencies;`
+	_, err := d.client.Exec(ctx, sql)
+	return err
+}
+
+func (d *Deps) Group(ctx context.Context, group string) (deps.Group, error) {
+	var result deps.Group
+	result.ID = group
+
+	sql := `SELECT dependency_id, COALESCE(resolved, false) FROM InstantGroupDependencies gd 
+				LEFT JOIN InstantDependencies d ON d.id = gd.dependency_id
+				WHERE gd.group_id = $1`
+	records, err := d.client.Query(ctx, sql, group)
+	if err != nil {
+		return result, err
+	}
+	defer records.Close()
+
+	for records.Next() {
+		var dep deps.Dependency
+		if err := records.Scan(&dep.ID, &dep.Resolved); err != nil {
+			return result, err
+		}
+		result.Dependencies = append(result.Dependencies, dep)
+	}
+
+	return result, nil
+}
+
+// Make generates a new dependency (but it does not store the information about it in the database
+//
+// Generation algorithm: UUID
+func (d *Deps) Make(_ context.Context) (deps.Dependency, error) {
+	id := uuid.New().String()
+
+	return deps.Dependency{
+		ID:       id,
+		Resolved: false,
+	}, nil
+}
+
+// MakeGroup creates a new dependency group
+//
+// NOTE: the new group's id is generated via a UUID algorithm
+func (d *Deps) MakeGroup(ctx context.Context, ids ...string) (string, error) {
+	id := uuid.New().String()
```

Comment:
Let's consider the following scenario:
1. User creates N dependencies
2. Resolves them all
3. Creates a group consisting of any subset of those dependencies

What happens to that group? Seems like it will never be resolved...